chain/chunks/src/lib.rs

```diff
@@ -1089,78 +1092,81 @@ impl ShardsManager {
         )
     }
 
+    /// Constructs a vector of [`ReceiptsProof`]s for specified chunk.
+    fn make_outgoing_receipts_proofs<T>(
+        &self,
+        chunk_header: &ShardChunkHeader,
+        outgoing_receipts: &Vec<Receipt>,
+        make_elem: impl Fn(ReceiptProof) -> T,
```

Comment:
I'd expect that we'd return vector of ReceiptProof, but here I see that we have this 'make_elem' instead. Can you explain in the comment why? is it due to performance reasons?